### PR TITLE
New Port: Cat Warrior

### PR DESCRIPTION
**Pull request by JanTrueno**
# New Port for Cat Warrior

## Game Information
- **Title**: Cat Warrior
- **URL**: https://store.steampowered.com/app/2640520/Cat_Warrior/

### CFW Tests
Ensure your game has been tested on all major CFWs:
- [X] AmberELEC
- [X] ArkOS
- [X] ROCKNIX
- [X] muOS

### Resolution Tests
Test all major resolutions:
- [X] 480x320
- [X] 640x480
- [] 720x720 (RGB30)
- [X] Higher resolutions (e.g., 1280x720)

## File Structure
- Your port should have the following structure:
  - portname/
    - port.json
    - README.md
    - screenshot.jpg
    - cover.jpg
    - gameinfo.xml
    - Port Name.sh
    - portname/
      - <portfiles here>

## Additional Resources
For an in-depth guide on creating a pull request, refer to: [PortMaster Game Packaging Guide](https://portmaster.games/packaging.html#creating-a-pull-request)

### CODE REVIEW

File: ports/catwarrior/Cat Warrior.sh [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+export LD_LIBRARY_PATH="$GAMEDIR/libs:/usr/config/emuelec/lib32:/usr/lib32:$LD_LIBRARY_PATH"
+export PATH="$monodir/bin:$PATH"
+
+export FNA3D_FORCE_DRIVER=OpenGL
+export FNA3D_OPENGL_FORCE_ES3=1
+export FNA3D_OPENGL_FORCE_VBO_DISCARD=1
+
+$GPTOKEYB "mono" &
+pm_platform_helper "mono"
+$TASKSET mono tiny_slash.exe 2>&1 | tee /dev/tty0 $GAMEDIR/log.txt
```
Comment (Cebion) [inline ports/catwarrior/Cat Warrior.sh:53]: old logging?

**Review by Cebion — DISMISSED** [review]

File: ports/catwarrior/Cat Warrior.sh (hunk 1 above)
Comment (JanTrueno) [reply ports/catwarrior/Cat Warrior.sh:53]: Yeah this is what johnny used on Celeste. I'll update this later today.

File: ports/catwarrior/Cat Warrior.sh [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+monofile="$controlfolder/libs/mono-6.12.0.122-aarch64.squashfs"
+$ESUDO mkdir -p "$monodir"
+$ESUDO umount "$monofile" || true
+$ESUDO mount "$monofile" "$monodir"
+
+# Setup savedir linux
+$ESUDO rm -rf ~/.local/share/tiny_slash
+mkdir -p ~/.local/share
+mkdir -p "$GAMEDIR/savedata"
+ln -sfv "$GAMEDIR/savedata" ~/.local/share/tiny_slash
```
Comment (cdeletre) [inline ports/catwarrior/Cat Warrior.sh:36]: Linking is done with `bind_directories` in the next line, just keep this line
```suggestion
mkdir -p "$GAMEDIR/savedata"
```

File: ports/catwarrior/port.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    "genres": [
+      "action",
+      "platformer"
+    ],
+    "image": null,
+    "rtr": false,
+    "exp": false,
+    "runtime": "mono-6.12.0.122-aarch64.squashfs",
+    "reqs": [],
+    "arch": []
```
Comment (cdeletre) [inline ports/catwarrior/port.json:27]: Not as important as missing the armhf tag but still :)
```suggestion
    "arch": [
        "aarch64"
    ]
```

**Review by cdeletre — CHANGES_REQUESTED** [review]
(=^.^=)

File: ports/catwarrior/port.json (hunk 3 above)
Comment (kloptops) [reply ports/catwarrior/port.json:27]: No arch if you use a runtime. :D

File: ports/catwarrior/port.json [hunk 4]
```diff
@@ -24,6 +24,8 @@
     "exp": false,
     "runtime": "mono-6.12.0.122-aarch64.squashfs",
     "reqs": [],
-    "arch": []
+    "arch": [
+        "aarch64"
```
Comment (kloptops) [inline ports/catwarrior/port.json:28]:
```suggestion
```

Runtime overrides arch. 😁

**Review by kloptops — CHANGES_REQUESTED** [review]

**Review by Cebion — APPROVED** [review]

**Review by cdeletre — APPROVED** [review]

**Pull request merged**